src/kv/MemDB.cc

```diff
@@ -397,7 +401,7 @@ void MemDB::MDBWholeSpaceIteratorImpl::fill_current()
 
 bool MemDB::MDBWholeSpaceIteratorImpl::valid()
 {
-  if (m_key_value.first.empty()) {
+  if (!(*is_valid) || m_key_value.first.empty()) {
```

Comment:
Rest looks fine, But I am not sure if current iterator becomes invalid then we should return invalid or restart iterator from current position. And if we return invalid, will bluestore take care of this and restart new iterator. I will confirm soon,